packages/common/src/type-helpers.ts

```diff
@@ -152,8 +179,11 @@ export function deepFreeze<T>(object: T): T {
 
     if (value && typeof value === 'object') {
       try {
-        deepFreeze(value);
+        deepFreeze(value, visited, `${path}.${name}`);
       } catch (err) {
+        if (typeof value !== 'object' || value.constructor.name !== 'Uint8Array') {
```

Comment:
You are checking that `if (value && typeof value === 'object') {` already
do you need to check again because deepFreeze could change it?
Also, a comment why Uint8Array is special may help